### PR TITLE
Traffic lights ZMQ API

### PR DESCRIPTION
**Pull request by pijaro**
## Types of PR

- [x] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

## Description

Adds ZMQ API for traffic lights control.

### CODE REVIEW

File: simulation/simulation_interface/include/simulation_interface/constants.hpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ const unsigned int despawn_entity = 5561;
 const unsigned int update_entity_status = 5562;
 const unsigned int attach_lidar_sensor = 5563;
 const unsigned int attach_detection_sensor = 5564;
+const unsigned int update_traffic_ligths = 5565;
```
Comment (hakuturu583) [inline simulation/simulation_interface/include/simulation_interface/constants.hpp:44]: Typo was detected.

File: simulation/simulation_interface/src/zmq_multi_server.cpp [hunk 2]
```diff
@@ -121,6 +127,8 @@ MultiServer::MultiServer(
     protocol, hostname, simulation_interface::ports::attach_lidar_sensor));
   attach_detection_sensor_sock_.bind(simulation_interface::getEndPoint(
     protocol, hostname, simulation_interface::ports::attach_detection_sensor));
+  update_traffic_lights_sock_.bind(simulation_interface::getEndPoint(
+    protocol, hostname, simulation_interface::ports::update_traffic_ligths));
```
Comment (hakuturu583) [inline simulation/simulation_interface/src/zmq_multi_server.cpp:131]: typo was detected

File: simulation/traffic_simulator/include/traffic_simulator/api/api.hpp [hunk 3]
```diff
@@ -86,7 +86,10 @@ class API
       simulation_interface::ports::attach_lidar_sensor),
     attach_detection_sensor_client_(
       simulation_interface::protocol, simulation_interface::HostName::LOCALHOST,
-      simulation_interface::ports::attach_detection_sensor)
+      simulation_interface::ports::attach_detection_sensor),
+    update_traffic_lights_client_(
+      simulation_interface::protocol, simulation_interface::HostName::LOCALHOST,
+      simulation_interface::ports::update_traffic_ligths)
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/api/api.hpp:92]: typo was detected

File: simulation/traffic_simulator/src/api/api.cpp [hunk 4]
```diff
@@ -311,6 +311,25 @@ bool API::updateSensorFrame()
...
 
+bool API::updateTrafficLightsInSim()
+{
+  simulation_api_schema::UpdateTrafficLightsRequest req;
+  simulation_api_schema::UpdateTrafficLightsResponse res;
+  if (entity_manager_ptr_->trafficLightsChanged()) {
+    auto ids = entity_manager_ptr_->getTrafficLightIds();
+    for (auto id : ids) {
+      simulation_api_schema::TrafficLightState state;
+      auto traf_light = entity_manager_ptr_->getTrafficLightInstance(id);
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/api/api.cpp:322]: This word was detected as typo.
Can you fix to traffic_light?

File: simulation/traffic_simulator/src/api/api.cpp [hunk 5]
```diff
@@ -311,6 +311,25 @@ bool API::updateSensorFrame()
...
+      simulation_interface::toProto(
+        static_cast<const autoware_perception_msgs::msg::TrafficLightState>(traf_light), state);
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/api/api.cpp:324]: This word was detected as typo.
Can you fix to traffic_light?

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Code looks good.
Can you fix typos?

**Comment by pijaro** [comment]
@hakuturu583 Typos should be fixed now. Sorry for that :bow:

**Review by hakuturu583 — APPROVED** [review]
Great work!
Thank you!

**Pull request merged**